sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionAsyncClient.java

```diff
@@ -0,0 +1,333 @@
+package com.azure.core.http.httpurlconnection;
+
+import com.azure.core.http.*;
+import com.azure.core.util.*;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import com.azure.core.http.HttpHeader;
+import reactor.core.scheduler.Schedulers;
+
+import java.io.*;
+import java.net.*;
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Duration;
+import java.util.*;
+
+/**
+ * This class provides a HttpUrlConnection implementation for the {@link HttpClient} interface.
+ * Creating an instance of this class can be achieved by using the {@link HttpUrlConnectionAsyncClientBuilder} class.
+ *
+ * @see HttpClient
+ * @see HttpUrlConnectionAsyncClientBuilder
+ */
+public class HttpUrlConnectionAsyncClient implements HttpClient {
+    private static final ClientLogger LOGGER = new ClientLogger(HttpUrlConnectionAsyncClient.class);
+    private final long connectionTimeout; // in milliseconds format for HttpUrlConnection methods
+    private final long readTimeout; // in milliseconds format for HttpUrlConnection methods
+    private final Duration writeTimeout;
+    private final Duration responseTimeout;
+    private final ProxyOptions proxyOptions;
+
+    HttpUrlConnectionAsyncClient(Duration connectionTimeout, Duration readTimeout, Duration writeTimeout,
+                                 Duration responseTimeout, ProxyOptions proxyOptions) {
+        this.connectionTimeout = connectionTimeout == null ? -1 : connectionTimeout.toMillis();
+        this.readTimeout = readTimeout == null ? -1 : readTimeout.toMillis();
+        this.writeTimeout = writeTimeout;
+        this.responseTimeout = responseTimeout;
+
+        this.proxyOptions = proxyOptions;
```

Comment:
Question for @alzimmermsft: What are the semantics of ProxyOptions here. The type is not settable, so in theory a user can modify the ProxyOptions passed in to the builder, and we therefore need to construct the appropriate strings in our requests every time. If the semantics should instead have been that this type was immutable, or that we copied on set, then we could cache the string that is concatenated every single time.

Reply:
In azure-core-http-netty and azure-core-http-okhttp the proxy is used by the client builder at build time of the HttpClient and any mutation after build shouldn't affect the created HttpClient. That said, there wasn't anything done to explicitly set a certain behavior, a test should be added to the shared HttpClient tests with a test that:

1. Sets a proxy on an HttpClient builder
2. Build an HttpClient instance
3. Change the ProxyOptions after build
4. Ensure the HttpClient instance isn't mutated implicitly after construction

For now, the ProxyOptions here should be known to be constant to align closer with immutability.